### PR TITLE
SAA-184 AM Events Endpoint: Incorporate activities (AM API)

### PR DESCRIPTION
**Pull request by markford-moj**
Altered ScheduledEventService to use activities management service for 'activities' if prison hasn't been rolled out

### CODE REVIEW

File: src/main/resources/application-local.yml [hunk 1]
```diff
@@ -30,9 +30,9 @@ system:
 # Container-provider local auth server
 hmpps:
   auth:
-    url: https://sign-in-dev.hmpps.service.justice.gov.uk/auth
+    url: http://localhost:8090/auth
 
 prison:
   api:
-    url: https://api-dev.prison.service.justice.gov.uk
+    url: http://localhost:8091
```
[the remark this answers is not in the document]
Comment (markford-moj) [reply src/main/resources/application-local.yml]: Will set this back to dev

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceTest.kt [hunk 2]
```diff
@@ -3,22 +3,105 @@ package uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.service
...
+        900001, dateRange
+      )
+    ).thenReturn(courtHearingsMono)
+
+    val result = service.getScheduledEventsByDateRange(
+      "MDI", "A11111A", dateRange
+    )
+
+    verify(scheduledInstanceService, times(1)).getActivityScheduleInstancesByDateRange(any(), any(), any())
+    verify(prisonApiClient, times(0)).getScheduledActivities(any(), any())
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceTest.kt:84]: Line 84.  No need for `times(0)` can just do 

` verify(prisonApiClient, never().getScheduledActivities(any(), any())`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceTest.kt [hunk 3]
```diff
@@ -63,6 +157,9 @@ class ScheduledEventServiceTest {
       LocalDateRange(LocalDate.of(2022, 10, 1), LocalDate.of(2022, 11, 5))
     )
 
+    verify(scheduledInstanceService, times(0)).getActivityScheduleInstancesByDateRange(any(), any(), any())
+    verify(prisonApiClient, times(1)).getScheduledActivities(any(), any())
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceTest.kt:161]: Default is `times(1`) anyway for verify, don't really need it.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceTest.kt [hunk 4]
```diff
@@ -3,22 +3,105 @@ package uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.service
...
+      prisonApiClient.getScheduledCourtHearings(
+        900001, dateRange
+      )
+    ).thenReturn(courtHearingsMono)
+
+    val result = service.getScheduledEventsByDateRange(
+      "MDI", "A11111A", dateRange
+    )
+
+    verify(scheduledInstanceService, times(1)).getActivityScheduleInstancesByDateRange(any(), any(), any())
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceTest.kt:83]: `times(1)` not needed.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM minor comments.

**Review by matt-crowson-moj — APPROVED** [review]

**Pull request merged**